ios/FluentUI/Bottom Sheet/CommandingItem.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// An object representing a command.
+///
+/// `CommandingItem` defines the high level properties and behavior of a command. Its visual representation is determined by
+/// the `BottomCommandingController`.
+@objc(MSFCommandingItem)
+open class CommandingItem: NSObject {
+
+    /// A closure that's called when the command is triggered
+    @objc open var action: (CommandingItem) -> Void
+
+    /// The title of the command item.
+    @objc open var title: String {
+        didSet {
+            if title != oldValue {
+                delegate?.commandingItem(self, didChangeTitleFrom: oldValue)
+            }
+        }
+    }
+
+    /// A `UIImage` to be displayed with the command.
+    @objc open var image: UIImage {
+        didSet {
+            if image != oldValue {
+                delegate?.commandingItem(self, didChangeImageFrom: oldValue)
+            }
+        }
+    }
+
+    /// A `UIImage` used when the command is represented as a button in selected state.
+    @objc open var selectedImage: UIImage? {
+        didSet {
+            if selectedImage != oldValue {
+                delegate?.commandingItem(self, didChangeSelectedImageFrom: oldValue)
+            }
+        }
+    }
+
+    /// Indicates whether the command is currently on.
+    ///
+    /// When `commandType` is set to `.toggle`, the item toggles this automatically before calling `action`.
+    @objc open var isOn: Bool {
```

Comment:
isOn is kinda like isSelected? As in the command is pressed?

Reply:
Yup, I chose isOn, because a command can be represented as a button or switch (and maybe more in the future), and isSelected really only makes sense in buttons. So this is meant to reflect the on/off state of the command, not the underlying view. The controller then interprets this however it wants, depending on what view this command is bound to.